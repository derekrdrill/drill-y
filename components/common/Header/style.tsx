import styled from 'styled-components';
import { AppBar, Grid } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const HeaderRoot = styled(AppBar)({
  backgroundColor: 'white !important',
  borderBottom: '1px solid #DFDFDF',
  boxShadow: 'none !important',
  zIndex: '3 !important',
});

export const HeaderTop = styled(Grid)({
  borderBottom: '1px solid #DFDFDF',
  padding: '5px 0',
});

export const HeaderLogo = styled.img({
  '&:hover': {
    opacity: 0.8,
  },
  cursor: 'pointer',
  height: 70,
  width: 100,
});

export const HeaderMenuText = styled.h4({
  '&:hover': {
    backgroundPositionX: '0%',
    backgroundSize: '100% 0.1em',
  },
  backgroundImage: 'linear-gradient(#212121, #212121)',
  backgroundPositionX: '100%',
  backgroundPositionY: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color: 'black',
  cursor: 'pointer',
  fontFamily: `'Josefin Sans', sans-serif !important`,
  fontWeight: '600 !important',
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
});

export const HeaderSearchIcon = styled(SearchIcon)({
  '&:hover': {
    color: '#a501c6',
    transform: 'scale(1.6)',
  },
  cursor: 'pointer',
  marginTop: 7,
  transition: 'all 5s ease-in-out',
  transform: 'scale(1.4)',
});
