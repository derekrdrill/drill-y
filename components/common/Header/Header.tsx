import * as React from 'react';
import { useRouter } from 'next/router';
import { Grid } from '@mui/material';

import { HeaderLogo, HeaderMenuText, HeaderRoot, HeaderSearchIcon, HeaderTop } from './style';

import Link from 'next/link';

const Header = () => {
  const router = useRouter();

  return (
    <HeaderRoot>
      <HeaderTop
        container
        justifyContent='space-around'
        sx={{ paddingLeft: { xs: '15px', sm: '20px' }, paddingRight: { xs: '15px', sm: '20px' } }}
      >
        <Grid item xs={4} lg={1} xl={2} sx={{ paddingTop: '40px' }} />
        <Grid item xs={4} lg={1} xl={2}>
          <Grid container display={{ lg: 'none' }}>
            <Grid item xs={12}>
              <Grid container justifyContent='center'>
                <Link href='/'>
                  <HeaderLogo src='/donutsImg.png' />
                </Link>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
        <Grid item md={1} xl={2} display={{ xs: 'none', lg: 'inline-block' }} />
      </HeaderTop>
      <Grid
        container
        sx={{ paddingLeft: { xs: '15px', sm: '20px' }, paddingRight: { xs: '15px', sm: '20px' } }}
      >
        <Grid item xs={12}>
          <Grid container justifyContent={{ xs: 'space-around', lg: 'space-between' }}>
            <Grid item display={{ xs: 'none', lg: 'inline-flex' }}>
              <Link href='/'>{/* <HeaderLogo src='/donutsImg.png' /> */}</Link>
            </Grid>
            <Grid item>
              <Grid
                columnSpacing={10}
                container
                justifyContent='space-around'
                sx={{ marginTop: 1.5 }}
              ></Grid>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </HeaderRoot>
  );
};

export default Header;
