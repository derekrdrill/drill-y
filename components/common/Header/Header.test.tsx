jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Header from './Header';

describe('Header unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(<Header isQuickMenuOpen={false} setIsQuickMenuOpen={jest.fn()} />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
