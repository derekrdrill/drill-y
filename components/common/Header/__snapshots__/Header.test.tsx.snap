// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header unit tests renders as expected 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed sc-jsMahE dJgdkS mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiGrid-root MuiGrid-container sc-kFuwaP cprlVZ css-eqlkfr-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-1 MuiGrid-grid-xl-2 css-2kga1j-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-1 MuiGrid-grid-xl-2 css-1e0qk4k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-wado3p-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <img
              className="sc-irTswW ddJncj"
              onClick={[Function]}
              src={
                {
                  "process": [Function],
                }
              }
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-beqWaB cIWvBF css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              />
              Order Now
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-10 MuiGrid-grid-xl-8 css-ytozq0-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-n302lm-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-zgfqgc-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-96gxtk-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-beqWaB eEtcVx css-sghohy-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Join Rewards
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium sc-beqWaB kNuXFO css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Sign In
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-fbJfA ealoin css-kinq8d-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            onClick={[Function]}
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 sc-fsQiph jlKuiZ css-1idn90j-MuiGrid-root"
            >
              <span
                className="MuiBadge-root sc-hLseeU gHjxGw css-1c32n2y-MuiBadge-root"
              >
                <span
                  className="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-fu0map-MuiBadge-badge"
                />
              </span>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-dmqHEX ItziR css-kinq8d-MuiSvgIcon-root"
                data-testid="ShoppingBagIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 6h-2c0-2.21-1.79-4-4-4S8 3.79 8 6H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-8 4c0 .55-.45 1-1 1s-1-.45-1-1V8h2v2zm2-6c1.1 0 2 .9 2 2h-4c0-1.1.9-2 2-2zm4 6c0 .55-.45 1-1 1s-1-.45-1-1V8h2v2z"
                />
              </svg>
              ;
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-1 MuiGrid-grid-xl-2 css-1j87gx2-MuiGrid-root"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-container css-seg8ra-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1gk8ux-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-zgfqgc-MuiGrid-root"
        >
          <img
            className="sc-irTswW ddJncj"
            onClick={[Function]}
            src={
              {
                "process": [Function],
              }
            }
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-19s77ke-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-162za67-MuiGrid-root"
            >
              <h4
                className="sc-csCMJt fJzAVp"
                color="textPrimary"
                onClick={[Function]}
              >
                Menu
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-oszuia-MuiGrid-root"
            >
              <h4
                className="sc-csCMJt fJzAVp"
                color="textPrimary"
                onClick={[Function]}
              >
                Menu
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <h4
                className="sc-csCMJt fJzAVp"
                color="textPrimary"
              >
                Reward
              </h4>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <h4
                className="sc-csCMJt fJzAVp"
                color="textPrimary"
              >
                Retail
              </h4>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-17maom6-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-e2ivvl-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-beqWaB jIWNKs css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              />
              Order Now
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
